src/helpers/spotify_manager.ts

```diff
@@ -458,133 +472,151 @@ export default class SpotifyManager {
 
     private generateSongMatchingPromise(
         song: SpotifyTrack,
-        isPremium: boolean
+        isPremium: boolean,
+        guildID: string
     ): Promise<QueriedSong | string> {
-        return new Promise(async (resolve) => {
-            const aliasIDs: Array<number> = [];
-            for (const artist of song.artists) {
-                const lowercaseArtist = normalizePunctuationInName(artist);
-                const artistMapping = State.artistToEntry[lowercaseArtist];
-                if (artistMapping) {
-                    aliasIDs.push(artistMapping.id);
-                    if (State.aliases.artist[lowercaseArtist]) {
-                        for (const alias of State.aliases.artist[
-                            lowercaseArtist
-                        ]) {
-                            const lowercaseAlias =
-                                normalizePunctuationInName(alias);
-
-                            if (lowercaseAlias in State.artistToEntry) {
-                                aliasIDs.push(
-                                    State.artistToEntry[lowercaseAlias].id
-                                );
+        return new Promise(async (resolve, reject) => {
+            try {
```

Comment:
If an exception is thrown in this fn (unlikely as it may be), it will be unhandled, and we won't remove the `guildsParseInProgress` lock. I can shrink it down to the scope of the db call. If you still think it's not useful, I can take it out